File: Muc_Do_2_Trung_Binh/Bai6_DemDoanConK/Solution.py
import sys
def solve():
    try:
        sys.stdin = open('BAI6.INP', 'r')
        sys.stdout = open('BAI6.OUT', 'w')
    except:
        pass
    data = sys.stdin.read().split()
    if not data: return
    n = int(data[0])
    k = int(data[1])
    arr = [int(x) for x in data[2:2+n]]
    cur = 0
    ans = 0
    cnt = {0: 1}
    for x in arr:
        cur += x
        ans += cnt.get(cur - k, 0)
        cnt[cur] = cnt.get(cur, 0) + 1
    print(ans)
solve()

File: Muc_Do_2_Trung_Binh/Bai6_DemDoanConK/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    if (fopen("BAI6.INP", "r")) {
        freopen("BAI6.INP", "r", stdin);
        freopen("BAI6.OUT", "w", stdout);
    }
    int n;
    long long k;
    if (!(cin >> n >> k)) return 0;
    long long cur = 0, ans = 0;
    map<long long, int> cnt;
    cnt[0] = 1;
    for (int i = 0; i < n; ++i) {
        long long x;
        cin >> x;
        cur += x;
        if (cnt.count(cur - k)) {
            ans += cnt[cur - k];
        }
        cnt[cur]++;
    }
    cout << ans << "\n";
    return 0;
}
